### ISSUE TITLE
Parameters of an instantiated new module are wrong: default values are reached with a delay

### ISSUE THREAD

**Issue by antoniotuzzi** [VCVRack/Rack#872]
(DEBUGGING on MAC)

I was investigating a crash happening into my module Model277. 

And I discovered this NEW behaviour.

The Model277 is a multi-tap delay and is driven by a PARAM named **PARAM_TIME**

>     addParam(ParamWidget::create<blackknob>(Vec(20, 292), module, Model277Module::PARAM_TIME, 0.005, 0.2, 0.1));

default should be 0.1 instantiating it

but I discovered that is always 0 (ZERO) at startup and after 896 sample the value starts to move
(kind there is a smoother or a lagger on the value!) and reaches the 0.1 value after 8530 samples !!

why params are not inited to the default immediately ? (the test was done many time, just create a new module from scratch)

THIS is happening ONLY when creating a NEW module from the MENU.
If I duplicate it, is in the correct conditions immediately

**Comment by mschack** [comment]
This may not be your issue, but I was getting funny start up values until I realized that I was still calling module->reset() at the end of my module constructor and forgot to change it to module->onReset().  Apparently module->reset() still compiles...

**Comment by antoniotuzzi** [comment]
OnReset internally still calls "reset" (so couldn't be the problem)

**Comment by AndrewBelt** [comment]
@mschack `reset()` is deprecated but will be removed later.
https://github.com/VCVRack/Rack/blob/v0.6/include/engine.hpp#L91-L94

@antoniotuzzi The Module is first created with all parameters set to 0. Later, the ModuleWidget is given control of the Module, and its values are set by each ParamWidget. If your module is crashing during this process, it needs range protection of some of its params.

**Comment by antoniotuzzi** [comment]
@AndrewBelt but why the value is not changed directly, and is smoothed ?
maybe we need a new event: "OnActivate" 
and only after OnActivate, the step function can be called

I explain: my default is 0.1  but my range is 0.005 to 1.0
so, I can put a guard between 0.005 and 1.0
but why I have to wait 8k samples to reach my default value ?

**Comment by AndrewBelt** [comment]
`onActivate()` would not be appropriate because knob smoothing isn't the responsibility of the Module.

If smoothing is enabled on a knob, it will call `engineSetParamSmooth()` once it is changed. I'd need a way to distinguish whether the knob is being changed from a `setDefaultValue()` call or a `setValue()` call. Can't think of a way to do that, but it's not a huge problem since every Module should be able to handle it.

**Comment by antoniotuzzi** [comment]
is it not a big problem, because is happening just when creating a new one
and after 180~200 msecs the module is stable

anyway was unjustified the start of the range from a value out of the range (0.0 instead of the 0.005)

I've guards now (for this module, but maybe I need for others!)

**Comment by AndrewBelt** [comment]
Duplicate of https://github.com/VCVRack/Rack/issues/650
